test/gcs/helper_container_test.go

```diff
@@ -0,0 +1,291 @@
+//go:build linux
+
+package gcs
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+
+	"github.com/containerd/containerd/namespaces"
+	ctrdoci "github.com/containerd/containerd/oci"
+	oci "github.com/opencontainers/runtime-spec/specs-go"
+
+	"github.com/Microsoft/hcsshim/internal/guest/prot"
+	"github.com/Microsoft/hcsshim/internal/guest/runtime"
+	"github.com/Microsoft/hcsshim/internal/guest/runtime/hcsv2"
+	"github.com/Microsoft/hcsshim/internal/guest/stdio"
+	"github.com/Microsoft/hcsshim/internal/guest/storage"
+	"github.com/Microsoft/hcsshim/internal/guest/storage/overlay"
+	"github.com/Microsoft/hcsshim/internal/guestpath"
+
+	testoci "github.com/Microsoft/hcsshim/test/internal/oci"
+)
+
+// todo: autogenerate/fuzz realistic specs
+
+//
+// testing helper functions for generic container management
+//
+
+const tailNull = "tail -f /dev/null"
+
+// will call unmountRootfs during cleanup
+func createStandaloneContainer(ctx context.Context, t testing.TB, host *hcsv2.Host, id string, extra ...ctrdoci.SpecOpts) *hcsv2.Container {
+	ctx = namespaces.WithNamespace(ctx, testoci.DefaultNamespace)
+	scratch, rootfs := mountRootfs(ctx, t, host, id)
+	// spec is passed in from containerd and then updated in internal\hcsoci\create.go:CreateContainer()
+	opts := testoci.DefaultLinuxSpecOpts(id,
+		ctrdoci.WithRootFSPath(rootfs),
+		ctrdoci.WithProcessArgs("/bin/sh", "-c", tailNull),
+	)
+	opts = append(opts, extra...)
+	s := testoci.CreateLinuxSpec(ctx, t, id, opts...)
+	r := &prot.VMHostedContainerSettingsV2{
+		OCIBundlePath:    scratch,
+		OCISpecification: s,
+	}
+
+	t.Cleanup(func() {
+		unmountRootfs(ctx, t, scratch)
+		// hcsv2.RemoveNetworkNamespace(ctx, id)
```

Comment:
remove?

Reply:
yeah, its unneeded for standalone containers